### DISCUSSION TITLE
Numerical simulations don't show anisotropic features.

### DISCUSSION THREAD

**Discussion by Johnney12138** [idaholab/moose#23521] [Q&A Modules: Phase field] [answered]
Hello,
I was trying to solve a phase field problem. After running, the numerical simulations don't show anisotropic features, but the file has the block about anisotropy. I don't know if it's the reason for the parameter, or do you have any suggestions for this?
The complete code is attached as follows:
```

[Mesh]
    type = GeneratedMesh
    dim = 2               
    nx = 300               # Number of elements in the X direction.
    ny = 300               # Number of elements in the Y direction.
    xmin = 0
    xmax = 5              # Upper X Coordinate of the generated mesh.
    ymin = 0
    ymax = 5
    uniform_refine = 2
[]

[Variables]
    [./w]       # concentration
    [../]
    [./phi]   # phase field
    [../]
    [./T]   # temperature
    [../]
[]

[ICs]
    [./w]
      type = ConstantIC
      variable = w
      value = 0.3
    [../]
    [./T]
      type = ConstantIC
      variable = T
      value = 2.1
    [../]
    [./phi]
      type = SmoothCircleIC
      profile = TANH
      variable = phi
      x1 = 2.5
      y1 = 2.5
      invalue = -1
      outvalue = 1
      radius = 1
      int_width = 0.405578
    [../]
[]

[Kernels]
  # phase field
    [./phi_dot]
      type = SusceptibilityTimeDerivative
      variable = phi
      f_name = tau
    [../]
    [./AllenCahn]
      type = AllenCahn
      variable = phi
      f_name = Fr
    [../]  
    [./anisoACinterface1]
      type = ACInterfaceKobayashi1
      variable = phi
    [../]
    [./anisoACinterface2]
      type = ACInterfaceKobayashi2
      variable = phi
    [../]
  # Chemical potential.
    [./w_dot]
      type = SusceptibilityTimeDerivative   
      variable = w
      f_name = chi                          # Susceptibility function F defined in a FunctionMaterial.
      coupled_variables = 'phi'
    [../]
    [./Diffusion]
      type = MatDiffusion                   # Diffusion equation Kernel that takes an isotropic Diffusivity from a material property
      variable = w
      diffusivity = Dchi                    # D*chi defined in [material] > [mobility]
    [../]
    [./coupled_etaa0dot]
      type = CoupledSusceptibilityTimeDerivative
      variable = w
      v = phi
      f_name = ft
      coupled_variables = 'phi'
    [../]
    [./coupled_antitrapping]
      type = AntitrappingCurrent
      variable = w
      v = phi
      f_name = rhodiff
    [../]
    # temperature.
    [./T_dot]
      type = TimeDerivative
      variable = T
    [../]
    [./coupled_CoefDiffusion]
      type = CoefDiffusion
      variable = T
      coef = 1e8
    [../]
    [./phi_dot_T]
      type = CoefCoupledTimeDerivative
      variable = T
      v = phi
      coef = 2
    [../]    
[]
  
[Materials]
    [./material]
      type = InterfaceOrientationMaterial
      op = phi
      anisotropy_strength = 0.05
      mode_number = 4
      eps_bar = 0.01
    [../]
    [./speexpress]
      type = ParsedMaterial           
      coupled_variables = 'phi w T'
      f_name = Fr
      material_property_names = 'lambda MCi'
      function = 'phi^3-phi+lambda*(1-phi^2)^2*(MCi*w+T)'
    [../]
    [./mobility]
      type = ParsedMaterial
      coupled_variables = 'w'  
      f_name = tau
      material_property_names = 'tau0 Le MCi k eps'
      expression = 'tau0*eps^2*(1/Le+MCi*(1+(1-k)*w))' 
    [../]
    [./chemp]
      type = ParsedMaterial
      coupled_variables = 'phi'
      f_name = chi
      material_property_names = 'k'
      function = '1+k-(1-k)*phi' 
    [../]
    [./matdiff]
      type = ParsedMaterial
      coupled_variables = 'phi'
      f_name = Dchi
      material_property_names = 'D'
      function = 'D*(1-phi)' 
    [../]
    [./antitrap]
      type = ParsedMaterial
      coupled_variables = 'w'
      f_name = rhodiff
      material_property_names = 'k wid'
      function = 'wid/sqrt(2)*(1+(1-k)*w)' 
    [../]
    [./int]
      type = ParsedMaterial
      coupled_variables = 'w'
      f_name = ft
      material_property_names = 'k'
      function = '1+(1-k)*w' 
    [../]
    [./const]
      type = GenericConstantMaterial
      prop_names =  'tau0   Le  MCi   k  lambda   D  wid L'
      prop_values = '1e-5  1.0  0.15  0.15  3.1913  2.0  0.1 1'
    [../]
[]

  
[Preconditioning]
    [./SMP]
      type = SMP        
      full = true       
    [../]
[]
  
[Executioner]
    type = Transient
    scheme = bdf2           # second-order backward differentiation formula
    solve_type = PJFNK      # NEWTON JFNK PJFNK
    petsc_options_iname = '-pc_type -pc_hypre_type -ksp_gmres_restart'
    petsc_options_value = 'hypre    boomeramg      31'
    l_tol = 1.0e-2          # linear tolerance
    l_max_its = 30          # max linear iterations
    nl_max_its = 100         # max non-linear iterations
    nl_rel_tol = 1.0e-4     # non-linear relative tolerance
    nl_abs_tol = 1.0e-4       # non-linear absolute tolerance
    end_time = 0.1          # The end time of the simulation. Default:1e+30
    line_search = none
    [./TimeStepper]
      type = IterationAdaptiveDT  # Adjust the timestep based on the number of iterations
      optimal_iterations = 6
      iteration_window = 2
      dt = 0.0005                 # The timestep size between solves
      cutback_factor = 0.75        # Factor to apply to timestep if difficult convergence
      growth_factor = 1.1         # Factor to apply to timestep if easy convergence
    [../]
[]

[Adaptivity]
  initial_steps = 3                   # The number of adaptive steps to do based on the initial condition.
  max_h_level = 3                     # Maximum number of times a single element can be refined. If 0 then infinite.
[]
  

  
[Outputs]
    exodus = true
    interval = 50     # only output every 10 timesteps.
[]

[Debug]
  show_material_props = true
  show_var_residual_norms = true
[]
 ```

Thanks!

**Comment by laagesen** [comment, marked as answer]
Hello,

To help us understand what you are seeing, what is the physical system that you are trying to simulate? (i.e. pure material, alloy, isothermal or varying temperature, etc.) Are you trying to implement a model that is already published? If yes, can you tell us what paper it is based on? If no, can you write up a description of the governing equations that you are trying to implement? 

You may already be aware of this, but there is an input file that is one of the examples in the MOOSE repository that you can look at. This is for growth of a dendritic structure that has a snowflake-like morphology. It is for solidification of a pure material. This is a useful reference for you to start from: `moose/modules/phase_field/examples/anisotropic_interfaces/snow.i`

**Reply by Johnney12138** [reply]
Hello @laagesen ,
Thank you for your reply.
The physical system is about alloy (link: https://journals.aps.org/pre/abstract/10.1103/PhysRevE.69.051607). In this model (eqs. 45, 46, 47), in addition to the anisotropy of the interface energy, the mobility is also anisotropic. I looked carefully at the reference (snow.i) you mentioned. I guess because I put the mobility on the left side of the equal sign, numerical simulations don't show anisotropic features. Do you have any suggestions for this?

**Reply by laagesen** [reply]
I would try to do some more basic troubleshooting of your model first before worrying about the anisotropy. If you have the sign of the mobility incorrect it will cause more problems than just not having anisotropy appear. If the mobility has the wrong sign, you may see a nucleus shrink rather than grow. I would first start with a 1D test of a supersaturated liquid with a solid region in one side of the domain, in conditions where you expect the solid to grow, in that case the size of the solid region should grow. Then do a 2D problem without anisotropy, with a circular solid seed in the center of the simulation domain. In that case you should see the circular seed grow while maintaining the circular shape, at least until it gets to be comparable in size to the simulation domain volume. Once you make sure the model is working for those simpler cases, then try turning on the anisotropy.

**Reply by Johnney12138** [reply]
Thank you very much.
I will refer to your valuable suggestions. I would like to ask you one more question, how to choose the value of the radius in the initial condition (SmoothCircleIC)?

**Reply by laagesen** [reply]
The value is not critical, I would just make sure to use something significantly larger than your interface width but significantly smaller than your simulation domain.

**Reply by Johnney12138** [reply]
Thank you very much for your response, which has been very helpful to me. I also appreciate @GiudGiud for organizing the code. Thank you both for your assistance.

**Reply by Johnney12138** [reply]
Hello @laagesen ,
I'm sorry to bother you again.
How should we determine the value of the free energy barrier height?
Thank you very much.

**Reply by laagesen** [reply]
From Eq. (49) in the Phys Rev E paper, you should use the interfacial energy gamma for the physical system you are working with, then for your chosen interfacial width W, you can solve for the free energy barrier height H.

**Reply by Johnney12138** [reply]
Thank you very much for your reply.
